mlir/lib/Dialect/SCF/Transforms/LoopPipelining.cpp

```diff
@@ -156,6 +160,11 @@ bool LoopPipelinerInternal::initializeLoopInfo(
     }
   }
 
+  if (!verifySchedule()) {
+    LDBG("--invalid schedule: " << op << " -> BAIL");
+    return false;
+  }
```

Comment:
nit: should the implementation of `verifySchedule` not be defined above the function that contains this check (above `LoopPipelinerInternal`?) Most of the MLIR codebase and this file `LoopPipelining.cpp` do this.

Reply:
I don't think there is such a rule but having  it closer to the other functions doing pre-condition checks makes sense. I moved it there.